src/utils.cpp

```diff
@@ -1071,7 +1071,8 @@ std::string hex_format_buffer(const char *buf,
                               bool escape_hex)
 {
   // Allow enough space for every byte to be sanitized in the form "\x00"
-  char s[size * 4 + 1];
+  std::vector<char> str(size * 4 + 1);
+  char *s = str.data();
```

Comment:
```suggestion
  std::string s(size * 4 + 1, '\0');
```

Can we just make this one a std::string from the beginning? Then there's no need to copy all the data again at the end

Reply:
I had the same thought (no, really) and just wanted to make sure you are paying attention. :cat: 
Fixed in the latest commit.